### ISSUE TITLE
[Bug]: SSHD container starts only once

### ISSUE THREAD

**Issue by akkie** [testcontainers/testcontainers-dotnet#1152]
### Testcontainers version

3.8.0

### Using the latest Testcontainers version?

Yes

### Host OS

MacOS

### Host arch

ARM

### .NET version

6.0.402

### Docker version

```shell
Client:
 Cloud integration: v1.0.35+desktop.13
 Version:           26.0.0
 API version:       1.45
 Go version:        go1.21.8
 Git commit:        2ae903e
 Built:             Wed Mar 20 15:14:46 2024
 OS/Arch:           darwin/arm64
 Context:           desktop-linux

Server: Docker Desktop 4.29.0 (145265)
 Engine:
  Version:          26.0.0
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       8b79278
  Built:            Wed Mar 20 15:18:02 2024
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### Docker info

```shell
Client:
 Version:    26.0.0
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.13.1-desktop.1
    Path:     /Users/christian/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.26.1-desktop.1
    Path:     /Users/christian/.docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container. (Docker Inc.)
    Version:  0.0.27
    Path:     /Users/christian/.docker/cli-plugins/docker-debug
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /Users/christian/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.23
    Path:     /Users/christian/.docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     /Users/christian/.docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.1.0
    Path:     /Users/christian/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/christian/.docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.6.3
    Path:     /Users/christian/.docker/cli-plugins/docker-scout
WARNING: Plugin "/Users/christian/.docker/cli-plugins/docker-scan" is not valid: failed to fetch metadata: fork/exec /Users/christian/.docker/cli-plugins/docker-scan: no such file or directory

Server:
 Containers: 20
  Running: 14
  Paused: 0
  Stopped: 6
 Images: 42
 Server Version: 26.0.0
 Storage Driver: stargz
  driver-type: io.containerd.snapshotter.v1
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.6.22-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 10
 Total Memory: 17.54GiB
 Name: docker-desktop
 ID: 723d0312-f721-48b2-aca5-04e97053fb73
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///Users/christian/Library/Containers/com.docker.docker/Data/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### What happened?

I try to implement parallel running integration tests with Gherkin. To have an isolated test environment. The app and the containers are instantiated in the `[BeforeScenario]` step and shutdown in the `[AfterScenario]` step. During instantiation I also setup the port forwarding from the container to the app. Because they run in parallel, each started app has a different port. For each test run I call `TestcontainersSettings.ExposeHostPortsAsync` with the actual app port.

Now the first test is successful, the others not. If I look at the started containers, I can see that the SSHD container was only started once with the port from the first run.

![image](https://github.com/testcontainers/testcontainers-dotnet/assets/307006/297cde1a-5224-4923-9287-f3c4dd537ad2)

**Comment by HofmeisterAn** [comment]
> I can see that the SSHD container was only started once with the port from the first run.

IIRC, this is by design. The port forwarding container is implemented as a singleton. I did some tests, and I cannot reproduce the issue. The randomly opened host ports are successfully forwarded.

https://github.com/testcontainers/testcontainers-dotnet/assets/9199345/cf6f6f9a-705c-4624-acf6-b439fa37cb33

**Comment by akkie** [comment]
This means your logic is the following?

-----> parallel
-> get free app port
-> start app with free port
-> call ExposeHostPortsAsync with free app port
-> start containers
<-----

Or do you call the ExposeHostPortsAsync with the app ports upfront? Because the method accepts multiple ports?

**Comment by akkie** [comment]
Yes, you are right. The forwarding for random generated ports works. I can see in my Dapr logs, that it could connect to the application. It seems that I have another issue, when running things in parallel. So I will close the issue. Thanks for your support!
